### DISCUSSION TITLE
Fooocus style selector, styles management UI ideas

### DISCUSSION THREAD

**Discussion by wisdombox44** [lllyasviel/Fooocus#2714] [Ideas]
Hi I have tried some UI features that would be so nice enable disable some style sets.
I have no knowledge  about coding i don't know is it posibble to build them.
Fell free to share your ideas about styles menu management, thank you
Please see the atached files
1. Styles Menu Selector
![styles menu copy](https://github.com/lllyasviel/Fooocus/assets/123548315/bdc4e3e0-537f-44cc-abaf-bb39ff1f1f13)
2. Styles Menu Manager 
![styles menu 3 copy](https://github.com/lllyasviel/Fooocus/assets/123548315/231b98d9-b554-4432-8f92-f7e091b4805e)

**Comment by bsdesigner** [comment]
I've been meaning to develop a style management interface for a few days now, and now I'm very glad you've done it for me ))))) And seriously, this issue has been a long time coming! There are more and more style collections and it's getting more and more difficult to manage them....
As for your suggestion, I think that the first option is much more correct and easier to manage:
![Снимок экрана 2024-04-06 110825](https://github.com/lllyasviel/Fooocus/assets/36730003/d2bed2fa-d05f-4278-a0b3-2427c86d1900)

**Comment by mashb1t** [comment]
@wisdombox44 this is possible, but enabling all styles will exceed the technical capabilities of max prompt length (~5000 token). Theoretically this could only be implenented in JS, but it's probably easier to adjust the existing logic in Python.
I prsonally don't see that much of benefit to be honest. 

@bsdesigner the shown changes have, if even, only been implemented in UI and not in logic, i assume.

**Reply by bsdesigner** [reply]
Nowadays, in order not to overload Fooocus with styles, I leave in the "sdxl_styles" folder only the styles I plan to work with today, and change them the next day as needed. But as you can see, this is not very convenient and after changing the styles in the folder, you have to restart Fooocus all the time. In most cases, when working, I don't use styles at all, as all necessary data can be written in positive and negative prompts. But nevertheless, in time I plan to create accurate and detailed styles for art, design, architecture, photography and film. And of course, it would be very desirable to have an optimal system for managing them. In the meantime, we can change them manually...

**Comment by wisdombox44** [comment]
Thank you masb1t , if we have an option to enable disable some style sets without moving them from folder that would be so nice.
I ment to enable style sets that has been disabled. I would not try to create 1 images with 100 styles selected. Besides fooocus default V2, Fooocus enhance and Foocus sharp if you add more than 3 there is no benefit and does not make sense. The best results comes when only one style selected or mixture of 2 styles.

**Comment by Rogala** [comment]
This is not a software solution, but just a picture, I’m not a programmer, I can do it according to a template if I understand it.
This is my vision of the styles window, the standard Fooocus styles are always displayed, new users are guided by them, more advanced ones can add other sets of styles that appear below.
![style](https://github.com/lllyasviel/Fooocus/assets/7437582/76b2551a-4a13-425d-aca3-8e88766aded2)

**Comment by ucscode** [comment]
I had the same issue with style management. 
Fooocus styles are too many and knowing which to choose from is quite tiring.
So I created a UI that displays over 20 preview per each style which has made it easier for me to select any desired style more efficiently.
I was using it locally for a while because the size was too large but later I decided to share it online.
Hopefully it solves someone else problem.

You can check it our [here](https://w2l.click/fooocus).

---

![fooocus preview](https://i.ibb.co/DD0J4R6R/Fooocus-Styles-Samples-Ucscode-11-27-2025-01-32-PM-2.jpg)
